src/worker/plugins/run.ts

```diff
@@ -3,6 +3,14 @@ import { PluginEvent } from '@posthog/plugin-scaffold'
 import { PluginConfig, PluginsServer, PluginTaskType } from '../../types'
 import { processError } from '../../utils/db/error'
 
+export class IllegalOperationError extends Error {
+    name = 'IllegalOperationError'
+
+    constructor(operation: string) {
```

Comment:
not even sure it's a nit, but I don't think you need the constructor if it just calls super. The constructor of the parent class should be called anyway if this is omitted. Fine if you like the explicitness though.